engine/schema/src/main/java/com/cloud/upgrade/SystemVmTemplateRegistration.java

```diff
@@ -458,10 +466,32 @@ private Long createTemplateObjectInDB(SystemVMTemplateDetails details) {
         template.setState(VirtualMachineTemplate.State.Inactive);
         template.setDeployAsIs(Hypervisor.HypervisorType.VMware.equals(details.getHypervisorType()));
         template = vmTemplateDao.persist(template);
-        if (template == null) {
-            return null;
+        return template;
+    }
+
+    private VMTemplateZoneVO createOrUpdateTemplateZoneEntry(long zoneId, long templateId) {
+        VMTemplateZoneVO templateZoneVO = vmTemplateZoneDao.findByZoneTemplate(zoneId, templateId);
+        if (templateZoneVO == null) {
+            templateZoneVO = new VMTemplateZoneVO(zoneId, templateId, new java.util.Date());
+            templateZoneVO = vmTemplateZoneDao.persist(templateZoneVO);
+        } else {
+            templateZoneVO.setRemoved(GenericDaoBase.DATE_TO_NULL);
+            templateZoneVO.setLastUpdated(new java.util.Date());
+            if (vmTemplateZoneDao.update(templateZoneVO.getId(), templateZoneVO)) {
+                templateZoneVO = null;
+            }
+        }
+        return templateZoneVO;
```

Comment:
this returned _templateZoneVO_ is unused, and only being validated for null. I think, it is better through the exception when the update fails.

Reply:
Don't think it affects functionality

Reply:
> Don't think it affects functionality

correct, doesn't affect functionality. it's about the implementation. returning VMTemplateZoneVO is not required in this method.